src/cdmtaskserviceclient/client.py

```diff
@@ -389,7 +398,14 @@ def wait_for_completion(
                     self._client._log.info(
                         f"Job {self.id} transitioned from {job_state} to {state}"
                     )
-                if state in ("complete", "error"):
+                if state == "error":
+                    return res
+                elif wait_for_event_importer:
+                    # if either of the event completion keys are in the admin_meta dict
+                    if _EVENT_COMPLETION_KEYS & res["admin_meta"].keys():
+                        return res
+                    # otherwise keep polling even if job is complete
+                elif state == "complete":
```

Comment:
Why is `status` either `error` or `complete`? Can it be something else or None?

Reply:
Because those are the 2 possible terminal states of a job. Anything else means the job is still running, so the method needs to continue to wait. 

It's never None